CONTRIBUTING.md

```diff
@@ -0,0 +1,101 @@
+# Contributing
+
+## System Requirements
+
+Java 18 is recommended for the tooling, plugins, etc. Maven 3.8+ is recommended.
+
+## Compilation target(s)
+
+As in the Java-SDK, we target Java 8. The parent POM configures this automatically.
+
+## Adding a module
+
+1. Create a [standard directory structure](https://maven.apache.org/guides/introduction/introduction-to-the-standard-directory-layout.html) in the appropriate folder (`hooks/`, `providers/`).
+1. Create a new `pom.xml` in the root of your new module. It must inherit from the parent POM, which implements the javadoc, testing, publishing, and other boilerplate. Be sure to add `<!--x-release-please-version -->` on the line specifying the module version, so our release tooling can update it (see sample pom below).
+1. Add the new package to `release-please-config.json`.
+1. Add the new module to the `<modules>...</modules>` section in the parent `pom.xml`.
+1. Add a `version.txt` file with a version matching that in your new `pom.xml`, e.g. `0.0.1`.
+2. If you care to release a pre 1.0.0 version, add the same version above to `.release-please-manifest.json`. Failing to do this will release a `1.0.0` initial release.
```

Comment:
```suggestion
1. If you care to release a pre 1.0.0 version, add the same version above to `.release-please-manifest.json`. Failing to do this will release a `1.0.0` initial release.
```